middle_end/flambda2/backend_intf/flambda2_backend_impl.ml

```diff
@@ -0,0 +1,128 @@
+(**************************************************************************)
+(*                                                                        *)
+(*                                 OCaml                                  *)
+(*                                                                        *)
+(*                       Pierre Chambart, OCamlPro                        *)
+(*           Mark Shinwell and Leo White, Jane Street Europe              *)
+(*                                                                        *)
+(*   Copyright 2013--2021 OCamlPro SAS                                    *)
+(*   Copyright 2014--2021 Jane Street Group LLC                           *)
+(*                                                                        *)
+(*   All rights reserved.  This file is distributed under the terms of    *)
+(*   the GNU Lesser General Public License version 2.1, with the          *)
+(*   special exception on linking described in the file LICENSE.          *)
+(*                                                                        *)
+(**************************************************************************)
+
+[@@@ocaml.warning "+a-30-40-41-42"]
+
+module Flambda1_compilation_unit = Compilation_unit
+module Flambda1_linkage_name = Linkage_name
+
+module Compilation_unit = struct
+  include Flambda2_compilenv_deps.Compilation_unit
+
+  let of_flambda1_compilation_unit comp_unit =
+    let ident = Flambda1_compilation_unit.get_persistent_ident comp_unit in
+    let linkage_name =
+      comp_unit
+      |> Flambda1_compilation_unit.get_linkage_name
+      |> Flambda1_linkage_name.to_string
+      |> Flambda2_compilenv_deps.Linkage_name.create
+    in
+    create ident linkage_name
+end
+
+module Linkage_name = Flambda2_compilenv_deps.Linkage_name
+module Symbol = Flambda2_compilenv_deps.Symbol
+
+let symbol_for_module_block id =
+  assert (Ident.global id);
+  assert (not (Ident.is_predef id));
+  let comp_unit =
+    Compilenv.unit_for_global id
+    |> Compilation_unit.of_flambda1_compilation_unit
+  in
+  Symbol.unsafe_create
+    comp_unit
+    (Linkage_name.create (Compilenv.symbol_for_global id))
+
+let symbol_for_global' ?comp_unit id =
+  if Ident.global id && not (Ident.is_predef id) then
+    symbol_for_module_block id
+  else
+    let comp_unit =
+      match comp_unit with
+      | Some comp_unit -> comp_unit
+      | None ->
+        if Ident.is_predef id then Compilation_unit.predefined_exception ()
+        else Compilation_unit.get_current_exn ()
+    in
+    Symbol.unsafe_create
+      comp_unit
+      (Linkage_name.create (Compilenv.symbol_for_global id))
+
+let closure_symbol _ = failwith "Not yet implemented"
```

Comment:
just to confirm: it's one of the "wires" to plug in later.

Reply:
This is actually not used at all, so removed.